### PR TITLE
[JBIDE-23015] - Creating a route should have a default port

### PR DESCRIPTION
**Pull request by jeffmaury**
- Extends Route model

Signed-off-by: Jeff MAURY <jmaury@redhat.com>

### CODE REVIEW

File: src/main/java/com/openshift/internal/restclient/model/Route.java [hunk 1]
```diff
@@ -152,4 +173,31 @@ public void setDestinationCertificate(String destinationCertificate) {
 		}
 
 	}
+	
+	private class TargetPort implements ITargetPort {
+
+	    private boolean isString = true;
```
Comment (adietish) [inline src/main/java/com/openshift/internal/restclient/model/Route.java]: "isString" seems unused?

File: src/test/java/com/openshift/internal/restclient/model/v1/RouteTest.java [hunk 2]
```diff
@@ -63,7 +64,42 @@ public void createTLSConfigWhenUndefined() {
 		assertEquals("", tls.getKey());
 	}
 	
-	@Test
+    @Test
+    public void getPortWhenUndefined() {
+        ModelNode node = ModelNode.fromJSONString(Samples.V1_ROUTE_WO_TLS.getContentAsString());
+        route = new Route(node, client, ResourcePropertiesRegistry.getInstance().get(VERSION, ResourceKind.ROUTE));
+        ITargetPort port = route.getPort();
+        assertNull(port);
```
Comment (adietish) [inline src/test/java/com/openshift/internal/restclient/model/v1/RouteTest.java]: Shouldn't we, in an additional test, also assert that the port, that's returned if you route.getPort().getTargetPort() is -1?

Comment (jeffmaury) [inline src/test/java/com/openshift/internal/restclient/model/v1/RouteTest.java]: Do you mean if you don't select a particular port. In that case, getPort() will return null

Comment (adietish) [inline src/test/java/com/openshift/internal/restclient/model/v1/RouteTest.java]: I see a Route#createPort(). If nothing further is done (which is most likely in user code, not included in this PR) I see a TargetPort being create that will return -1 if you dont set it a numeric port. Missing something?

**Comment by adietish** [comment]
+1, LGTM

**Comment by adietish** [comment]
[merge]

**Pull request merged**